File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void dfs(TreeNode* root, int level, vector<vector<int>> &result){
        if (root == NULL) {
            return;
        }
        
        if (result.size() == level) {
            result.push_back({});
        }
        
        result[level].push_back(root->val);
        
        dfs(root->left, level + 1, result);
        dfs(root->right, level + 1, result);
    }
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> result;
        dfs(root, 0, result);
        return result;
    }
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        # res = []
        # def dfs(root, level):
        #     if not root:
        #         return
        #     if level == len(res):
        #         res.append([])
        #     res[level].append(root.val)
        #     dfs(root.left, level + 1)
        #     dfs(root.right, level + 1)
        # dfs(root, 0)
        # return res
        result = []
        queue = deque()
        if root:
            queue.append(root)
        while queue:
            res = []
            for i in range(len(queue)):
                node = queue.popleft()
                res.append(node.val)
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
            result.append(res)
        return result